src/Run/Setup.cs

```diff
@@ -77,40 +142,92 @@ private string FindSettingType(string valueToFind)
             return null;
         }
 
-        public void prepareValues(string os, Dictionary<string, string> parameters, string configFile)
+        public int PrepareValues(string os, Dictionary<string, string> parameters, string configFile)
         {
             SettingParameters = new Dictionary<string, string>(parameters);
             Os = os;
             ConfigurationFilePath = configFile;
+            // Add defaults for run tool settings if they haven't been defined in config.json
+            SetRunToolSettingsDefaults();
+
+            // Validate Settings before parsing out ToolSettings
+            int returnCode = ValidateSettings();
```

Comment:
My thought was that the benefit of completing the "PrepareValues" code outweighed the perf gain by failing faster here.